bind/flag.go

```diff
@@ -279,17 +279,25 @@ func HTTPServerConfig(fs *pflag.FlagSet, cfg *forwarder.HTTPServerConfig, prefix
 	fs.VarP(anyflag.NewValueWithRedact[*url.Userinfo](cfg.BasicAuth, &cfg.BasicAuth, forwarder.ParseUserinfo, RedactUserinfo),
 		namePrefix+"basic-auth", "", "<username[:password]>"+
 			"Basic authentication credentials to protect the server. ")
+}
 
-	httpLogModes := []httplog.Mode{
-		httplog.None,
-		httplog.ShortURL,
-		httplog.URL,
-		httplog.Headers,
-		httplog.Body,
-		httplog.Errors,
+func HTTPLogConfig(fs *pflag.FlagSet, logcfg *httplog.Config) {
+	valid := make([]string, 0, len(*logcfg))
+	for _, m := range *logcfg {
+		if m.Name != "" {
+			valid = append(valid, m.Name)
+		}
 	}
```

Comment:
The string building should happen in this for loop. We do not need to rewrite the slice.